File: frontend/src/types/parser.ts
export interface ASTSymbol {
  id: string;
  file_id: string;
  file_path?: string;
  symbol_type: 'function' | 'class' | 'import' | string;
  name: string;
  start_line: number;
  end_line: number;
  signature?: string;
  docstring?: string;
}

export interface ParseRepositorySummary {
  repository_id: string;
  repository_name: string;
  files_parsed: number;
  total_symbols_extracted: number;
}

export interface SymbolListResponse {
  total: number;
  symbols: ASTSymbol[];
}

File: backend/app/schemas/parser.py
from typing import Optional, List
from pydantic import BaseModel, Field, ConfigDict


class ASTSymbolResponse(BaseModel):
    id: str
    file_id: str
    file_path: Optional[str] = None
    symbol_type: str
    name: str
    start_line: int
    end_line: int
    signature: Optional[str] = None
    docstring: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ParseRepositorySummary(BaseModel):
    repository_id: str
    repository_name: str
    files_parsed: int
    total_symbols_extracted: int


class SymbolListResponse(BaseModel):
    total: int
    symbols: List[ASTSymbolResponse]
